flink-runtime/src/main/java/org/apache/flink/runtime/checkpoint/CheckpointRequestDecider.java

```diff
@@ -136,6 +136,11 @@
 				.map(unused -> queuedRequests.pollFirst());
 		}
 
+		CheckpointTriggerRequest first = queuedRequests.first();
```

Comment:
maybe we can reuse the `first` in `onTooEarly` function

Reply:
thanks, @klion26 , I will reuse it.

Reply:
Hi, @klion26  @rkhachatryan  I remove `onTooEarly` function and put all statements in `chooseRequestToExecute`.